types/parameter_types/abstract_parameter_type.go

```diff
@@ -0,0 +1,23 @@
+package parameter_types
+
+// LeafIntParamAbstractType represents an integer parameter for a parameterized type
+// Example: VARCHAR(L1) -> L1 is an LeafIntParamAbstractType
+// DECIMAL<P, 0> --> P Is an LeafIntParamAbstractType
+type LeafIntParamAbstractType string
```

Comment:
is there a corresponding non-`Leaf` param type? If not, do we really need to call it a "Leaf" type?

Is `AbstractType` what it is called in the substrait spec? Would `IntParamVar` be a better name if not?

Reply:
renamed it to ConcreteIntParam